openmdao/core/system.py

```diff
@@ -2940,7 +2940,10 @@ def get_design_vars(self, recurse=True, get_sizes=True, use_prom_ivc=True):
 
                 if 'size' not in meta:
                     if src_name in abs2idx:
-                        meta['size'] = sizes[owning_rank[src_name], abs2idx[src_name]]
+                        if meta['distributed']:
+                            meta['size'] = abs2meta[src_name]['global_size']
```

Comment:
Ken pointed me to justin's issue that is the basis for this PR, and based on that, this line should probably be `meta['size'] = sizes[model.comm.rank, abs2idx[src_name]]`